soti/message.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from dataclasses import dataclass, field
+from utils.constants import CmdID, NodeID
+from session_logger import parse_msg_body
+
+@dataclass
+class Message:
+    source: str
+    priority: int
+    sender: NodeID
+    recipient: NodeID
+    cmd_id: CmdID
+    body: bytes = field(default_factory=([0] * 7))
+    time: datetime = field(default_factory=
+        lambda:datetime.now().strftime("%T")
+    )
+
+    @classmethod
+    def deserialize(cls, source, msg_bytes: bytes) -> "Message":
```

Comment:
I think it would be better if source wasn't in the parameter list here. I'd say pass `None` into the constructor, and let the user set `msg.source` after it's created.

Reply:
Sure. I went for a default field for source so it's easier to debug using the logs, check 8c0b35c and let me know if that's good.

Reply:
Looks good!